tests/stepDefinitions/loginContext.js

```diff
@@ -1,26 +1,39 @@
+const {expect} =  require('@playwright/test');
 const { Given, When, Then } = require("@cucumber/cucumber");
 
-Given('the user {string} has navigated to the login page', function (string) {
-  // Write code here that turns the phrase above into concrete actions
-  console.log("solved1");
-});
+const loginBtnLocator  = ".btn-block"
+const emailFieldLocator = "#spree_user_email"
+const passwordFieldLocator = "#spree_user_password"
+const paraFieldLocator = ".mb-3"
+const errorLocator = ".alert-danger"
 
-When('user {string} logs in with email {string} and password {string}', function (user, email, password) {
-  // Write code here that turns the phrase above into concrete actions
-  console.log(user, email, password)
+Given('the user {string} has navigated to the login page',  async function (string) {
+  await page.goto("http://localhost:3000/admin/login");
+  expect(page.url()).toBe("http://localhost:3000/admin/login");
 });
 
-Then('user {string} should be navigated to the admin panel dashboard', function (string) {
-  // Write code here that turns the phrase above into concrete actions
-  console.log("solved2")
+When('user {string} logs in with email {string} and password {string}', async function (user, email, password) {
+  await page.fill(emailFieldLocator, email);
+  await page.fill(passwordFieldLocator, password);
+  await page.click(loginBtnLocator);
 });
 
-When('user {string} tries to log in with following details:', function (string, dataTable) {
-  // Write code here that turns the phrase above into concrete actions
-  console.log("solved")
+Then('user {string} should be navigated to the admin panel dashboard', async function (string) {
+await page.waitForURL("http://localhost:3000/", { timeout: 5000 });
```

Comment:
Please indent your code accordingly
```suggestion
    await page.waitForURL("http://localhost:3000/", { timeout: 5000 });
```